.tito/lib/git_annex_spec_builder.py

```diff
@@ -0,0 +1,168 @@
+import os
+import shutil
+from distutils.version import LooseVersion as loose_version
+from pkg_resources import require
+
+from tito.compat import getstatusoutput
+from tito.config_object import ConfigObject
+from tito.builder import GitAnnexBuilder
+from tito.builder.main import BuilderBase
+from tito.common import error_out, debug, run_command, get_spec_version_and_release, find_spec_like_file, \
+                        get_relative_project_dir, get_relative_project_dir_cwd
+
+
+class GitAnnexSpecBuilder(GitAnnexBuilder):
+
+    def __init__(self, name=None, tag=None, build_dir=None,
+            config=None, user_config=None,
+            args=None, **kwargs):
+
+        """
+        name - Package name that is being built.
+
+        version - Version and release being built.
+
+        tag - The git tag being built.
+
+        build_dir - Temporary build directory where we can safely work.
+
+        config - Merged configuration. (global plus package specific)
+
+        user_config - User configuration from ~/.titorc.
+
+        args - Optional arguments specific to each builder. Can be passed
+        in explicitly by user on the CLI, or via a release target config
+        entry. Only for things which vary on invocations of the builder,
+        avoid using these if possible.  *Given in the format of a dictionary
+        of lists.*
+        """
+        ConfigObject.__init__(self, config=config)
+        BuilderBase.__init__(self, name=name, build_dir=build_dir, config=config,
+                user_config=user_config, args=args, **kwargs)
+        self.build_tag = tag
+
+        self.build_version = self._get_build_version()
+        self.git_commit_id = self.build_version
+
+        if kwargs and 'options' in kwargs:
+            warn_out("'options' no longer a supported builder constructor argument.")
+
+        if self.config.has_section("requirements"):
```

Comment:
If this is a ConfigParser this if is redundant:
```
In [1]: from ConfigParser import ConfigParser

In [2]: c = ConfigParser()

In [3]: c.has_option('section', 'option')
Out[3]: False
```

Reply:
Sorry, I don't entirely follow the logic here.

Reply:
```has_option``` also works if there is no such section. Since that's on the next line it's redundant to check if the section exists.